lib/api.js

```diff
@@ -0,0 +1,26 @@
+async function request({ data, url, method }) {
+  const options = data
+    ? {
+        method,
+        body: JSON.stringify(data),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    : { method };
+  const response = await fetch(url, options);
+  const json = await response.json();
+  return json;
+}
+
+export async function createNewEvent(data) {
```

Comment:
I did not see you using this function in your code. Please clear up your code and delete everything you don't need right now.